### PR TITLE
test: use `T.TempDir` to create temporary test directory

### PR DESCRIPTION
**Pull request by Juneezee**
## Related Issues
A testing enhancement.

## Proposed Changes

Replaces `ioutil.TempDir` with `T.TempDir` function from the `testing` package to create temporary directory. The directory created by `T.TempDir` is automatically removed when the test and all its subtests complete. 

## Additional Info
Reference: https://pkg.go.dev/testing#T.TempDir

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] CI is green

### CODE REVIEW

File: extern/sector-storage/manager_test.go [hunk 1]
```diff
@@ -45,8 +45,7 @@ func (t testStorage) DiskUsage(path string) (int64, error) {
 }
 
 func newTestStorage(t *testing.T) *testStorage {
-	tp, err := ioutil.TempDir(os.TempDir(), "sector-storage-test-")
-	require.NoError(t, err)
+	tp := t.TempDir()
```
Comment (ZenGround0) [inline extern/sector-storage/manager_test.go:48]: This needs to be left as a ioutil temp dir to respect the `LOTUS_TEST_NO_CLEANUP` env var that was introduced for debugging

**Review by ZenGround0 — CHANGES_REQUESTED** [review]
Other than the test storage in the sector storage manager this LGTM.  It looks like you've been keeping the ioutil temp dirs around in places where the current test does not delete the dir 👍. However I am less familiar with these other tests so it's hard to be sure.

Please go through and double check there are no more places where this change will remove test directories the current tests wouldn't and then this will be good to go.

**Review by ZenGround0 — APPROVED** [review]

File: node/repo/fsrepo_test.go [hunk 2]
```diff
@@ -22,18 +17,15 @@ func genFsRepo(t *testing.T) (*FsRepo, func()) {
...
-		_ = os.RemoveAll(path)
-	}
+	return repo
 }
 
 func TestFsBasic(t *testing.T) {
-	repo, closer := genFsRepo(t)
-	defer closer()
-	//stm: @NODE_FS_REPO_LOCK_001,@NODE_FS_REPO_LOCK_002,@NODE_FS_REPO_UNLOCK_001
+  //stm: @NODE_FS_REPO_LOCK_001,@NODE_FS_REPO_LOCK_002,@NODE_FS_REPO_UNLOCK_001
```
Comment (magik6k) [inline node/repo/fsrepo_test.go:24]:
```suggestion
	//stm: @NODE_FS_REPO_LOCK_001,@NODE_FS_REPO_LOCK_002,@NODE_FS_REPO_UNLOCK_001
```

File: lib/backupds/backupds_test.go [hunk 3]
```diff
@@ -59,10 +58,8 @@ func TestNoLogRestore(t *testing.T) {
 }
 
 func TestLogRestore(t *testing.T) {
-	//stm: @OTHER_DATASTORE_RESTORE_001
-	logdir, err := ioutil.TempDir("", "backupds-test-")
-	require.NoError(t, err)
-	defer os.RemoveAll(logdir) // nolint
+  //stm: @OTHER_DATASTORE_RESTORE_001
```
Comment (magik6k) [inline lib/backupds/backupds_test.go:61]:
```suggestion
	//stm: @OTHER_DATASTORE_RESTORE_001
```

**Review by magik6k — APPROVED** [review]
Thanks!

**Pull request merged**